src/python/pants/backend/python/typecheck/mypy/mypyc.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.python.goals.setup_py import DistBuildEnvironment, DistBuildEnvironmentRequest
+from pants.backend.python.target_types import PythonDistribution
+from pants.backend.python.typecheck.mypy.subsystem import (
+    MyPy,
+    MyPyConfigFile,
+    MyPyFirstPartyPlugins,
+)
+from pants.backend.python.util_rules import pex_from_targets
+from pants.backend.python.util_rules.pex import Pex, PexRequest
+from pants.backend.python.util_rules.pex_from_targets import RequirementsPexRequest
+from pants.backend.python.util_rules.pex_requirements import PexRequirements
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import BoolField, Target
+from pants.engine.unions import UnionRule
+
+
+class UsesMyPycField(BoolField):
+    alias = "uses_mypyc"
+    default = False
+    help = "If true, this distribution is built using mypyc."
```

Comment:
What else would be relevant to a user? For example

* do we expect a hand-written setup.py / pyproject.toml? If so, with what content?
* clarifying it uses the version of `mypy` from `[mypy].version`

Reply:
Added more detail.